pcs/lib/resource_agent/xml.py

```diff
@@ -137,20 +138,36 @@ def load_fake_agent_metadata(
 
 
 def parse_metadata(
-    name: ResourceAgentName, metadata: _Element
+    name: ResourceAgentName,
+    metadata: _Element,
+    report_processor: reports.ReportProcessor,
+    report_warnings=True,
```

Comment:
I don't like the idea of passing 2 extra arguments to this function. This just doesn't seems to be the best place to solve this. I would prefer to keep this low level function simple (not dealing with reports).

As we don't want to always report this warning (I guess it's due to not outputting warnings to stderr), we should handle this case in higher layers. Returning real  ocf version of source metadata as another return value should enable us to make this check somewhere else and only in when its needed.

Reply:
I didn't like it either :)